ingestion_server/ingestion_server/ingest.py

```diff
@@ -319,6 +319,9 @@ def reload_upstream(table, progress=None, finish_time=None, approach="advanced")
             f"_Next: re-applying indices & constraints_"
         )
 
+    # The server sometimes hangs on or before this next step. Adding this debug
+    # message here because apparently that unblocks it? May have to do with GC.
+    log.debug("Getting downstream cursor again")
```

Comment:
Could try verifying this using `gc.collect`?

Reply:
I'm guessing that any additional call made here, prior to the database connection call, would unblock this step. Work a shot though :woman_shrugging:

Reply:
Hmmm... any additional call... would that include like `int("0")` or something? I wonder what just calling a function could be causing it to clear 🤔

Reply:
I'm not sure! What I noticed is that all of the following would cause code execution to continue as expected:
- `log.info`
- `log.debug`
- `print`

I'm find leaving in the GC and seeing if it pops up again. It's frustrating because it's so intermittent!

Reply:
Interesting. I wonder if it's somehow IO related?

Reply:
Maybe! I'm really at a loss